src/CommandAndStateMessageParser.cpp

```diff
@@ -0,0 +1,317 @@
+
+    return motor;
+}
+
+Grabber CommandAndStateMessageParser::getGrabberMotorInfo(string address)
+{
+    Grabber grabber;
+    MotorDiagnostics motor;
+    motor.current =
+        mJData["devices"][address]["grabber"]["motorDiagnostics"]["current"].asDouble();
+    motor.overcurrent =
+        mJData["devices"][address]["grabber"]["motorDiagnostics"]["overcurrent"].asBool();
+    vector<float> vector_motor;
+    vector_motor.push_back(
+        mJData["devices"][address]["grabber"]["motorDiagnostics"]["rpm"].asFloat()
+    );
+    motor.motor = samples::Joints::Raw(vector_motor);
+    grabber.motor_diagnostic = motor;
+    grabber.open = mJData["devices"][address]["grabber"]["openClose"].asDouble();
+    vector<float> vector_rotate;
+    vector_rotate.push_back(mJData["devices"][address]["grabber"]["rotate"].asFloat());
+    grabber.rotate_command = samples::Joints::Raw(vector_rotate);
+
+    return grabber;
+}
+
+TiltCameraHead CommandAndStateMessageParser::getCameraHeadInfo(string address)
+{
+    TiltCameraHead camera_head;
+    camera_head.light = mJData["devices"][address]["cameraHead"]["lights"].asDouble();
+    camera_head.laser = mJData["devices"][address]["cameraHead"]["lasers"].asBool();
+    vector<float> tilt_command;
+    tilt_command.push_back(
+        mJData["devices"][address]["cameraHead"]["tilt"]["speed"].asFloat()
+    );
+    camera_head.tilt_command = samples::Joints::Speeds(tilt_command);
+    MotorDiagnostics motor;
+    motor.current =
+        mJData["devices"][address]["cameraHead"]["tiltMotorDiagnostics"]["current"]
+            .asDouble();
+    motor.overcurrent =
+        mJData["devices"][address]["cameraHead"]["tiltMotorDiagnostics"]["overcurrent"]
+            .asBool();
+    vector<float> vector_motor;
+    vector_motor.push_back(
+        mJData["devices"][address]["cameraHead"]["tiltMotorDiagnostics"]["rpm"].asFloat()
+    );
+    motor.motor = samples::Joints::Raw(vector_motor);
+    camera_head.tilt_motor_diagnostics = motor;
+    camera_head.camera.brightness =
+        mJData["devices"][address]["cameraHead"]["camera"]["brightness"].asDouble();
+    camera_head.camera.exposure =
+        mJData["devices"][address]["cameraHead"]["camera"]["exposure"]["value"].asDouble(
+        );
+    camera_head.camera.focus =
+        mJData["devices"][address]["cameraHead"]["camera"]["focus"]["value"].asDouble();
+    camera_head.camera.saturation =
+        mJData["devices"][address]["cameraHead"]["camera"]["saturation"].asDouble();
+    camera_head.camera.sharpness =
+        mJData["devices"][address]["cameraHead"]["camera"]["sharpness"].asDouble();
+    camera_head.camera.zoom.ratio =
+        mJData["devices"][address]["cameraHead"]["camera"]["zoom"]["ratio"].asDouble();
+    vector<float> vector_speed;
+    vector_speed.push_back(
+        mJData["devices"][address]["cameraHead"]["camera"]["zoom"]["speed"].asFloat()
+    );
+    camera_head.camera.zoom.speed = samples::Joints::Speeds(vector_speed);
```

Comment:
In the case of the zoom, I think it's a bit too much to use a joint ;-). Just add zoomRatio and zoomSpeed as float to camera_head.camera. Remember to document their units (and convert percentage to [0, 1] if there are some.